From 365465811113ed5018ee4c8d5b078b41095387fc Mon Sep 17 00:00:00 2001
From: "Paul \"Joey\" Clark" <joeytwiddle@gmail.com>
Date: Fri, 27 Aug 2021 08:09:49 +0800
Subject: [PATCH 30/85] Add git submodule to the build docs

This confused me, and looking at the issues, it has confused other people as well. [1](https://github.com/shadowsocks/shadowsocks-libev/issues/2758)

I'm not sure if we can automate it though, e.g. in `autogen.sh`. Because some people might want to build with a modified version of the libraries.

Perhaps we could automatically fetch the submodules IF we see they are empty...

Anyway, for now, we can at least document this step.
---
 README.md | 8 ++++++++
 1 file changed, 8 insertions(+)

diff --git a/README.md b/README.md
index 30ddd85..ca155f3 100644
--- a/README.md
+++ b/README.md
@@ -69,6 +69,14 @@ sudo snap install shadowsocks-libev --edge
 
 * * *
 
+### Initialise the build environment
+
+This repository uses submodules, so you should pull them before you start, using:
+
+```bash
+git submodule update --init --recursive
+```
+
 ### Pre-build configure guide
 
 For a complete list of available configure-time option,
-- 
2.25.1

